src/agentlab/experiments/launch_exp.py

```diff
@@ -66,8 +71,9 @@ def run_experiments(
             with make_dask_client(n_worker=n_jobs):
                 execute_task_graph(exp_args_list)
         elif parallel_backend == "sequential":
-            for exp_args in exp_args_list:
-                exp_args.run()
+            for exp_args, wait_func in zip(exp_args_list, wait_funcs):
+                wait_and_run(exp_args, wait_func)
+                # exp_args.run()
```

Comment:
This section is a little dated; dask backend is no longer supported. I assume the wait_func is to wait for servers to be available? If so, you can add your wait function inside the chat_model_args's make_model() method. That's what I did [here](https://github.com/ServiceNow/ui-copilot/blob/agentlab-finetuning-sync/finetuning/src/finetuning/toolkit_utils/chat_api.py#L176)